spring-integration-core/src/main/java/org/springframework/integration/transaction/TransactionHandleMessageAdvice.java

```diff
@@ -46,12 +46,16 @@
 	public TransactionHandleMessageAdvice() {
 	}
 
-	public TransactionHandleMessageAdvice(PlatformTransactionManager ptm, Properties attributes) {
-		super(ptm, attributes);
+	public TransactionHandleMessageAdvice(TransactionManager transactionManager, Properties transactionAttributes) {
+		setTransactionManager(transactionManager);
+		setTransactionAttributes(transactionAttributes);
 	}
```

Comment:
Need `final` overrides of these (or a new method taking both) to call from a CTOR - even though Spring is in violation of that rule in the super class.

Reply:
Well, I think we can live with this for a while.
See here: https://github.com/spring-projects/spring-framework/issues/24612
When that is fixed, we can revert this change to rely on super ctors.